### ISSUE TITLE
Internationalization (I18N) enablement

### ISSUE THREAD

**Issue by amfred** [Call-for-Code-for-Racial-Justice/Open-Sentencing#15]
We woud like to be able to translate what users see in the future. Enabling the repos for translation is step 1 of that.

**Comment by RickPoleshuck** [comment]
This seems like a good first project. Does Github let you add dependencies? My wife might help with Spanish language translations, but that issue seems dependent on this one.

**Comment by amfred** [comment]
You're right - we at least need to get all of the English strings in one file first.  Issue #16 
Then it's not hard to create another copy of the file with the Spanish in it.  
The exact file format depends on the translation tool we choose; still open to suggestions.

Heh. Github Enterprise+Zenhub (which is what I'm used to) lets us add dependencies, but I don't see how to do it here. Anyone else know how to do that?

**Comment by amfred** [comment]
By the way, the intent of this issue #15 is to choose the open source translation library.  Then #16 is to put the English strings into the format that the library needs.  And #17 is for a translation of that file into Spanish (or any other language we can get).

**Comment by amfred** [comment]
We might well have to choose different translation libraries/plug-ins for different repos because they use different programming languages, but if we choose one for the UI to start with that uses a standard file format, maybe we can at least use the same file format for the other repos.

**Comment by RickPoleshuck** [comment]
I have been working with jHipster <https://www.jhipster.tech/>. One of 
the things I like about the framework is that they have made good 
choices on which technologies to use. Does it make sense to create a 
slack channel to discuss #15?

**Comment by RickPoleshuck** [comment]
I am very new. Excuse me if I ask stupid questions. Why would you have 
translation in anything other than the UI? Legal documents might need to 
be translated, but that isn't part of the code base.

**Comment by blueivywave** [comment]
It sounds like the intent of this task is to provide content for multiple languages. Correct?

If that is the case, what APIs are being considered for this, or are being evaluated?

**Comment by amfred** [comment]
The lower level components like the Bias Detection Engine might end up generating text too, so I could see us passing a language code to them and getting back translated text.  It's fine to start with the UI first since we know that needs it for sure.

**Comment by amfred** [comment]
> It sounds like the intent of this task is to provide content for multiple languages. Correct?
> 
> If that is the case, what APIs are being considered for this, or are being evaluated?

You mean like Google translate?

**Comment by RickPoleshuck** [comment]
Again, I am very new, but since you asked...

I would use @ngx-translate. As I said earlier, I like jHipster's choices.

Google translate might be useful on the backend.

**Comment by blueivywave** [comment]
> We might well have to choose different translation libraries/plug-ins for different repos because they use different programming languages, but if we choose one for the UI to start with that uses a standard file format, maybe we can at least use the same file format for the other repos.  

Have you looked into IBM's Language Translator? It instantly translates web content. Here is the link: https://www.ibm.com/watson/services/language-translator/

**Comment by RickPoleshuck** [comment]
This is from an Angular project I am toying with:
     html:

         <span jhiTranslate="game.title">Game</span>

     i18n:

         "game": {
             "title": "Game",
             "createLabel": "Create a new Game",

I believe that this is the type of translation needed for the front end 
UI. This kind of translation is all hard coded in various languages and 
very efficient.

Translations on the back end that need to be done dynamically need a 
different solution. The IBM language translator might be the optimum 
choice for that.

**Comment by amfred** [comment]
I recently learned that Angular has a built-in translation framework, so I think we should start there with the UI, because it's built in Angular. 
https://angular.io/guide/i18n

**Comment by RickPoleshuck** [comment]
This article, https://phrase.com/blog/posts/best-libraries-for-angular-i18n/ , says that the Angular builtin has caught up with NGX-Translate. I am happy with either.

**Comment by RickPoleshuck** [comment]
I would get started with the implementation, but I haven't fully set up my development environment. authentication.service.ts and client_id is still confusing to me. Does it make sense for me to ask for help?

**Comment by carleyreardon3** [comment]
Not sure if you are still looking for ideas on this, or if this would even be something you consider, but if you considered using machine translation for this and wanted to make sure the model works well on the type of language you're using (legal language, etc), you could try fine tuning a machine translation model on a set of english sentences and a small set of manually translated ones. HuggingFace has a ton of machine translation models that are fairly easy to implement -- here is the link for one that translates English to Arabic (for example): https://huggingface.co/Helsinki-NLP/opus-mt-en-ar

**Comment by demilolu** [comment]
@carleyreardon3 we're still looking at this. we're also doing some translation for our five fifths voter project. For that we're looking at Watson Natural Language Translator, but we're also thinking about how to crowdsource human translators/reviewers as well. For Hugging Face, i forget is it via API, I think the issue with BERT or similar transformer models is the computation required. 

We might need a unified view of how we deal with translation across all our projects :). cc @upkarlidder
